cloud-controller-manager/civo/loadbalancer.go

```diff
@@ -189,7 +146,89 @@ func (l *loadbalancer) UpdateLoadBalancer(ctx context.Context, clusterName strin
 	}
 	lbuc.Backends = backends
 
-	ulb, err := l.client.civoClient.UpdateLoadBalancer(civolb.ID, &lbuc)
+	if ip := getReservedIP(service); ip != "" {
+		rip, err := l.client.civoClient.FindIP(ip)
+		if err != nil {
+			klog.Errorf("Unable to find reserved IP, error: %v", err)
+			return nil, err
+		}
+
+		// this is so that we don't try to reassign the reserved IP to the loadbalancer
+		if rip.AssignedTo.ID != civolb.ID {
+			_, err = l.client.civoClient.AssignIP(rip.ID, civolb.ID, "loadbalancer")
+			if err != nil {
+				klog.Errorf("Unable to assign reserved IP, error: %v", err)
+				return nil, err
+			}
+		}
+	} else {
+		ips, err := l.client.civoClient.ListIPs()
+		if err != nil {
+			klog.Errorf("Unable to list IPs, error: %v", err)
+			return nil, err
+		}
+		ip := findIPWithLBID(ips.Items, civolb.ID)
+		if ip != nil {
```

Comment:
It's not clear why when we find the IP, we now try and Unassign it. Do we need a comment to explain this logic?